File: Regression/Polynomial Regression/Polynomial_Regression/Polynomial Regression.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, y)

from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 10)
X_poly = poly_reg.fit_transform(X)
lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly, y)

#Linear Graph
plt.scatter(X, y, color = '#61f248')
plt.plot(X, lin_reg.predict(X), color = '#9fac73')
plt.title('Truth or Bluff (Linear Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

#Polynomial Graph
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = '#2bdf9e')
plt.plot(X_grid, lin_reg_2.predict(poly_reg.fit_transform(X_grid)), color = '#18a9df')
plt.title('Truth or Bluff (Polynomial Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

lin_reg.predict([[6.5]])
lin_reg_2.predict(poly_reg.fit_transform([[6.5]]))

File: Regression/Polynomial Regression/Polynomial Regression.R
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

lin_reg = lm(formula = Salary ~ ., data = dataset)

dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
poly_reg = lm(formula = Salary ~ ., data = dataset)

ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = '#A17C32') +
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)), colour = '#6EBCEE') +
  ggtitle('Truth or Bluff (Linear Regression)') +
  xlab('Level') +
  ylab('Salary')

ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = '#A17C32') +
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)), colour = '#6EBCEE') +
  ggtitle('Truth or Bluff (Polynomial Regression)') +
  xlab('Level') +
  ylab('Salary')

y_pred = predict(lin_reg, data.frame(Level = 6.5))

y_pred = predict(poly_reg, data.frame(Level = 6.5, Level2 = 6.5^2, Level3 = 6.5^3, Level4 = 6.5^4))